File: app/src/main/java/com/komsi/maleshop/fragment/CartFragment.java
package com.komsi.maleshop.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.komsi.maleshop.adapter.CartAdapter;
import com.komsi.maleshop.model.Produk;
import com.komsi.maleshop.R;

import java.util.ArrayList;

public class CartFragment extends Fragment {

    RecyclerView recyclerView;
    CartAdapter adapter;
    ArrayList<Produk> list;
    TextView hargaProduk;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_cart, container, false);

        recyclerView = view.findViewById(R.id.rv_cart);
        hargaProduk = view.findViewById(R.id.tv_total);

        recyclerView.setHasFixedSize(true);
        list = new ArrayList<>();
        list.add(new Produk("1", "Tolliver-x", R.drawable.bg1, "400000", "-", 0.0, "-","-"));
        list.add(new Produk("2", "AutivFixin", R.drawable.bg2, "450000", "-", 0.0, "-","-"));
        list.add(new Produk("1", "Banana Republic", R.drawable.bg3, "300000", "-", 0.0, "-","-"));

        adapter = new CartAdapter(getActivity(), list,hargaProduk);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(adapter);

        return view;
    }
}

File: app/src/main/java/com/komsi/maleshop/ui/fragment/CartFragment.kt
package com.komsi.maleshop.ui.fragment

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.androidnetworking.AndroidNetworking
import com.androidnetworking.common.Priority
import com.androidnetworking.error.ANError
import com.androidnetworking.interfaces.JSONObjectRequestListener
import com.androidnetworking.interfaces.StringRequestListener
import com.komsi.maleshop.R
import com.komsi.maleshop.adapter.CartAdapter
import com.komsi.maleshop.adapter.ProductCallback
import com.komsi.maleshop.helper.ToolbarHelper
import com.komsi.maleshop.model.Cart
import com.komsi.maleshop.model.Produk
import com.komsi.maleshop.persistence.Credential
import com.komsi.maleshop.ui.activity.AlamatActivity
import com.komsi.maleshop.ui.activity.PaymentActivity
import com.komsi.maleshop.utils.ConstApi
import com.komsi.maleshop.utils.toaster
import com.komsi.maleshop.viewmodel.CartViewModel
import kotlinx.android.synthetic.main.fragment_cart.*
import org.json.JSONObject
import java.text.DecimalFormat
import java.text.NumberFormat

class CartFragment : Fragment(), ProductCallback, View.OnClickListener {
    private lateinit var adapter: CartAdapter
    private lateinit var rvCart: RecyclerView
    private lateinit var tvTotal: TextView
    private lateinit var dialogLoading: DialogFragmentLoading
    private lateinit var tvAddressShip: TextView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_cart, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        rvCart = view.findViewById(R.id.rv_cart) as RecyclerView
        tvTotal = view.findViewById(R.id.tv_total) as TextView
        tvAddressShip = view.findViewById(R.id.tvAddressShip) as TextView
        dialogLoading = DialogFragmentLoading()

        fetchProfileData()

        btn_open_alamat.setOnClickListener {
            val intent = Intent(activity, AlamatActivity::class.java)
            startActivity(intent)
        }

        btnOrder.setOnClickListener(this)
        fetchCart()

        btnOrder.setOnClickListener {
            val intent = Intent(activity, PaymentActivity::class.java)
            startActivity(intent)
        }

    }

    private fun fetchCart() {
        rvCart.itemAnimator = null
        rvCart.layoutManager = LinearLayoutManager(requireContext())
        adapter = CartAdapter(requireContext(), this)
        rvCart.adapter = adapter

        val cartViewModel = ViewModelProvider(
                requireActivity(),
                ViewModelProvider.NewInstanceFactory()
        ).get(CartViewModel::class.java)
        cartViewModel.setList(requireActivity())
        cartViewModel.getList().observe(requireActivity(), androidx.lifecycle.Observer { contents ->
            adapter.setData(contents)
            tvTotal.text = getTotalPrice(contents)
        })
    }

    private fun fetchProfileData() {
        AndroidNetworking.initialize(requireContext())
        AndroidNetworking.post(ConstApi.PROFILE.value)
                .addHeaders("Authorization", "Bearer ${Credential.getToken(requireContext())}")
                .setTag("profile")
                .setPriority(Priority.HIGH)
                .build()
                .getAsJSONObject(object : JSONObjectRequestListener {
                    override fun onResponse(response: JSONObject) {
                        Log.d("profileResponse", response.toString())
                        tvAddressShip.text = response.getString("alamat")

                    }

                    override fun onError(anError: ANError?) {
                        Log.d("profileResponse", anError.toString())
                    }
                })
    }

    fun getTotalPrice(list: ArrayList<Cart>): String {
        var price = 0.0
        list.map {
            price += it.harga * it.qty.toDouble()
        }
        return addCurrency(price)
    }
    fun addCurrency(price: Double): String{
        val formatter: NumberFormat = DecimalFormat("#,###")
        return "Rp ${formatter.format(price)}"
    }
    override fun onResume() {
        ToolbarHelper.setTitle(getString(R.string.cart))
        fetchProfileData()
        super.onResume()
    }

    override fun productOnClick(produk: Produk) {

    }

    override fun onFavoriteClick() {

    }

    override fun onClick(p0: View) {
        when (p0.id) {
            R.id.btnOrder -> {
                dialogLoading.show(requireFragmentManager(), "order")
                processOrder()
                fetchCart()
            }
        }
    }

    private fun processOrder() {
        AndroidNetworking.initialize(context)
        AndroidNetworking.post(ConstApi.ORDER.value)
                .addHeaders("Authorization", "Bearer ${Credential.getToken(requireContext())}")
                .setTag("order")
                .setPriority(Priority.HIGH)
                .build()
                .getAsString(object : StringRequestListener {
                    override fun onResponse(response: String?) {
                        "Success".toaster(requireContext())
                        dialogLoading.dismiss()
                    }

                    override fun onError(anError: ANError?) {
                        "Error".toaster(requireContext())
                        dialogLoading.dismiss()
                    }
                })
    }
}